### DISCUSSION TITLE
Ansible Manager Docker-type dependency extraction

### DISCUSSION THREAD

**Discussion by ckstevenson** [renovatebot/renovate#18190] [Request Help] [answered]
Hey all,

Following this [documentation](https://docs.renovatebot.com/modules/manager/ansible/), I am trying to simply add some variables to an ansible playbook and get a PR when a new version of that tool is released. I am able to see that my files are matched (`Matched 64 file(s) for manager ansible`), but I am having issues finding out how I should  declare the variable. I didn't care to do any fancy matching, I simply wanted to use what was the default. So I added some test variables to a file, such as:
```
hub: docker.io/hashicorp/terraform: 1.0.0
hub1: docker.io/hashicorp/terraform:1.0.0
terraform_version: 1.0.0
tf: hashicorp/terraform:1.0.0
tf2: hashicorp/terraform: 1.0.0
```
In this case, I am installing Terraform to a host, so I'd like to update my playbook when a new Terraform version is released. Hoping I could solve this with Renovate in this way rather than, say, using a command runner to source it from a `Dockerfile` which Renovate versions.

Thoughts? Unable to find it in the docs, probably because I can't see what's right in from of me.

**Comment by viceice** [comment, marked as answer]
you probably need to use the regex manager, as the Ansible manager only supports docker images, here is the regex used

https://github.com/renovatebot/renovate/blob/4a0314aa8144433834de417cf045b2854d64825d/lib/modules/manager/ansible/extract.ts#L14

**Comment by ckstevenson** [comment]
Ah thanks for linking the regex! This helped me verify what was possible and that I would have to use the regex manager to build upon this. I will try that next. Next time I'll have to get my hands dirty and check the source code :wink: . Much appreciated!

**Reply by ckstevenson** [reply]
For anybody who ends up here, I was able to solve my issue with a renovate.json file containing:

```
  "regexManagers": [
    {
      "fileMatch": ["(^|/)*.yml$"],
      "matchStrings": [
        "terraform_version:\\s*\"(?<currentValue>.*)\"\\s*"
      ],
      "depNameTemplate": "hashicorp/terraform",
      "datasourceTemplate": "github-releases"
    }
  ]
```

and an Ansible playbook with `terraform_version: "1.0.5"`. I am having issues with the optional quantifier `?`, which I have outlined in [this](https://github.com/renovatebot/renovate/discussions/18208) discussion.

**Reply by viceice** [reply]
your file match is missing a dot before the first star.
